components/datetime/src/provider/mod.rs

```diff
@@ -194,16 +195,90 @@ pub mod gregory {
             pub short: Cow<'static, str>,
         }
 
+        #[derive(Debug, PartialEq, Clone)]
+        #[cfg_attr(
+            feature = "provider_serde",
+            derive(serde::Serialize, serde::Deserialize)
+        )]
+        pub enum CountV1 {
+            Zero,
+            One,
+            Two,
+            Few,
+            Many,
+            Other,
+        }
+
+        #[derive(Debug, PartialEq, Clone)]
+        #[cfg_attr(
+            feature = "provider_serde",
+            derive(serde::Serialize, serde::Deserialize)
+        )]
+        pub struct SkeletonPatternV1 {
+            pub pattern: Cow<'static, str>,
+            #[cfg_attr(
+                feature = "provider_serde",
+                serde(skip_serializing_if = "Option::is_none")
```

Comment:
You need to also make this dependent on the serialize_none feature, which is necessary for bincode.  Find some boilerplate from elsewhere and copy it in here.

We should probably add a CI check that fails if bincode doesn't round-trip.